Backport fix for bb #1134 (cannot locate socket: path too long)

Use relative paths for socket

Index: sx-debian/server/test/run-nginx-test.sh
===================================================================
--- sx-debian.orig/server/test/run-nginx-test.sh
+++ sx-debian/server/test/run-nginx-test.sh
@@ -12,7 +12,8 @@ unset http_proxy
 unset HTTPS_PROXY
 unset https_proxy
 
-prefix=`mktemp -d $PWD/sx-test-XXXXXXXX`
+prefix_rel=`mktemp -d sx-test-XXXXXXXX`
+prefix="$PWD/$prefix_rel"
 cleanup () {
     (cd sxscripts && make clean)
     rm -rf $prefix
@@ -45,7 +46,7 @@ test -x "$built_nginx" && ln -s "$built_
 
 cp "$prefix/etc/sxserver/sxhttpd.conf.default" "$prefix/etc/sxserver/sxhttpd.conf"
 
-SXRUNDIR="$prefix/var/run/sxserver"
+SXRUNDIR="$prefix_rel/var/run/sxserver"
 SXSTOREDIR="$prefix/var/lib/sxserver"
 SXLOGFILE="$prefix/var/log/sxserver/sxfcgi.log"
 cat >"$prefix/etc/sxserver/sxfcgi.conf" <<EOF
@@ -63,9 +64,14 @@ fi
 cat >"$prefix/etc/sxserver/sxsetup.conf" <<EOF
 SX_NO_ROOT=1
 SX_RUN_DIR="$SXRUNDIR"
+SX_LOG_FILE="$SXLOGFILE"
+SX_LIB_DIR="$prefix/var/lib/sxserver"
 EOF
 
-sed -e "s|^user.*|user `whoami`;|" -e "s|listen .*443|listen 127.0.0.1:8443|g" -e "s|listen .*80|listen 127.0.0.1:8013|g" $prefix/etc/sxserver/sxhttpd.conf >$prefix/etc/sxserver/sxhttpd.conf.1
+sed -e "s|^user.*|user `whoami`;|" -e "s|listen .*443|listen 127.0.0.1:8443|g" \
+    -e "s|listen .*80|listen 127.0.0.1:8013|g" \
+    -e "s|unix:.*|unix:$SXRUNDIR/sxfcgi.socket;|" \
+    $prefix/etc/sxserver/sxhttpd.conf >$prefix/etc/sxserver/sxhttpd.conf.1
 mv "$prefix/etc/sxserver/sxhttpd.conf.1" "$prefix/etc/sxserver/sxhttpd.conf"
 
 "$prefix/sbin/sxadm" node --new --batch-mode "$SXSTOREDIR/data"
@@ -77,6 +83,7 @@ mv "$prefix/etc/sxserver/sxhttpd.conf.1"
 cleanup () {
     echo "cleaning up"
     "$prefix/sbin/sxserver" stop
+    cat "$SXLOGFILE"
     rm -rf $prefix
 }
 export SX_FCGI_OPTS="--config-file=$prefix/etc/sxserver/sxfcgi.conf"
@@ -86,14 +93,12 @@ trap cleanup EXIT INT
 
 "$prefix/bin/sxinit" --batch-mode --port=8013 --no-ssl --auth-file="$SXSTOREDIR/data/admin.key" --config-dir="$prefix/.sx" sx://localhost
 `dirname $0`/client-test --config-dir="$prefix/.sx" --filter-dir="`pwd`/../client/src/filters" sx://localhost || {
-    cat "$SXLOGFILE";
     cat $prefix/var/log/sxserver/sxhttpd-error.log;
     exit 1
 }
 
 perl `dirname $0`/fcgi-test.pl 127.0.0.1:8013 $SXSTOREDIR/data || {
     rc=$?
-    cat "$SXLOGFILE";
     cat $prefix/var/log/sxserver/sxhttpd-error.log;
     if [ "$rc" -eq 2 ]; then
         exit 77 # perl dep missing
